backend/src/main/java/com/bakdata/conquery/apiv1/FilterSearch.java

```diff
@@ -121,175 +98,108 @@ public List<TrieSearch<FEValue>> getSearchesFor(SelectFilter<?> filter) {
 	 */
 	public void updateSearch() {
 
-		jobManager.addSlowJob(new SimpleJob("Initialize Source Search", () -> {
+		jobManager.addSlowJob(new SimpleJob(
+				"Initialize Source Search",
+				() -> {
 
-			log.info("BEGIN loading SourceSearch");
+					log.info("BEGIN loading SourceSearch");
 
-			// collect all SelectFilters to the create searches for them
-			final List<SelectFilter<?>> allSelectFilters =
-					storage.getAllConcepts().stream()
-						   .flatMap(c -> c.getConnectors().stream())
-						   .flatMap(co -> co.collectAllFilters().stream())
-						   .filter(SelectFilter.class::isInstance)
-						   .map(f -> ((SelectFilter<?>) f))
-						   .collect(Collectors.toList());
+					// collect all SelectFilters to the create searches for them
+					final List<SelectFilter<?>> allSelectFilters =
+							storage.getAllConcepts().stream()
+								   .flatMap(c -> c.getConnectors().stream())
+								   .flatMap(co -> co.collectAllFilters().stream())
+								   .filter(SelectFilter.class::isInstance)
+								   .map(f -> ((SelectFilter<?>) f))
+								   .collect(Collectors.toList());
 
 
-			final Map<String, Stream<FEValue>> suppliers = new HashMap<>();
+					final Set<Searchable> suppliers = new HashSet<>();
 
-			// collect all tasks that are based on the filters configured label mappings
-			collectLabelTasks(allSelectFilters, suppliers);
+					// collect all tasks that are based on the filters configured label mappings
+					suppliers.addAll(collectLabelTasks(allSelectFilters));
 
-			// collect all tasks based on the filters optionally configured templates based on csvs
-			collectTemplateTasks(parserConfig, allSelectFilters, suppliers);
+					// collect all tasks based on the filters optionally configured templates based on csvs
+					suppliers.addAll(collectTemplateTasks(allSelectFilters));
 
-			// collect all tasks that are based on the raw data in the columns, these have no reference or template for mapping
-			collectColumnTasks(storage, allSelectFilters, suppliers);
+					// collect all tasks that are based on the raw data in the columns, these have no reference or template for mapping
+					suppliers.addAll(collectColumnTasks(allSelectFilters));
```

Comment:
Da das statisch ist von woher hier die Suchquellen kommen, kannst du auch ein Stream.concat machen, anstatt collectColumnTasks immer eine Set zu materialisieren

Reply:
Es ist aber sehr wahrscheinlich, dass das Duplikate enthalten wird, zB wenn eine Spalte mehrfach referenziert wird in Unterschiedlichen suchen, distinct brauche ich sie sowieso.